client/me/concierge/shared/calendar-page.js

```diff
@@ -56,54 +47,31 @@ const groupAvailableTimesByDate = ( availableTimes, timezone ) => {
 		.map( date => dates[ date ] );
 };
 
-class CalendarStep extends Component {
+class CalendarPage extends Component {
 	static propTypes = {
 		availableTimes: PropTypes.array.isRequired,
+		description: PropTypes.string.isRequired,
 		onBack: PropTypes.func.isRequired,
-		onComplete: PropTypes.func.isRequired,
+		onSubmit: PropTypes.func.isRequired,
 		site: PropTypes.object.isRequired,
 	};
 
-	onSubmit = timestamp => {
-		const { signupForm } = this.props;
-		const meta = {
-			message: signupForm.message,
-			timezone: signupForm.timezone,
-		};
-
-		this.props.bookConciergeAppointment(
-			WPCOM_CONCIERGE_SCHEDULE_ID,
-			timestamp,
-			this.props.currentUserId,
-			this.props.site.ID,
-			meta
-		);
-	};
-
-	componentWillUpdate( nextProps ) {
-		if ( nextProps.signupForm.status === CONCIERGE_STATUS_BOOKED ) {
-			// go to confirmation page if booking was successfull
-			this.props.onComplete();
-		}
-	}
-
 	render() {
-		const { availableTimes, currentUserLocale, onBack, signupForm, translate } = this.props;
+		const { availableTimes, currentUserLocale, description, disabled, onBack, onSubmit,
+			signupForm, title } = this.props;
 		const availability = groupAvailableTimesByDate( availableTimes, signupForm.timezone );
 
 		return (
 			<div>
-				<HeaderCake onClick={ onBack }>{ translate( 'Choose Concierge Session' ) }</HeaderCake>
-				<CompactCard>
-					{ translate( 'Please select a day to have your Concierge session.' ) }
-				</CompactCard>
+				{ onBack && <HeaderCake onClick={ onBack }>{ title }</HeaderCake> }
```

Comment:
Looking to other PRs, I can see that this conditional is here becase in the `Book` context you want a title bar with a back button, but in the `Reschedule` context you don't. How about instead, we just get rid of the title from this component and if an individual step wants to render a `HeaderCake` it can do so before it renders the `CalendarPage`?

That also opens up future instances where we might drop this in mid-page and want a bunch of stuff between the `HeaderCake` and the calendar step — we won't need messy conditionals in here, the parent component can just deal with that and drop this component in.

In that light, I'd also consider changing the component name from `CalendarPage` (which sounds like it should take up the whole screen) to maybe something like `AvailableTimePicker` (and `CalendarCard` could similarly be renamed like `AvailableTimeCard`).